src/Tickets/Admin/All_Tickets/List_Table.php

```diff
@@ -0,0 +1,257 @@
+<?php
+/**
+ * The list table for the All Tickets screen.
+ *
+ * @since  TBD
+ *
+ * @package TEC\Tickets\Admin
+ */
+
+namespace TEC\Tickets\Admin\All_Tickets;
+
+use WP_List_Table;
+
+/**
+ * Class List_Table.
+ *
+ * @since  TBD
+ *
+ * @package TEC\Tickets\Admin
+ */
+class List_Table extends WP_List_Table {
+	/**
+	 * The user option that will store how many attendees should be shown per page.
+	 *
+	 * @var string
+	 */
+	protected $per_page_option;
+
+	/**
+	 * The constructor.
+	 *
+	 * @since  TBD
+	 */
+	public function __construct() {
+		$screen = get_current_screen();
+
+		parent::__construct(
+			[
+				'singular' => 'ticket',
+				'plural'   => 'tickets',
+				'ajax'     => false,
+				'screen'   => $screen,
+			]
+		);
+
+		$this->per_page_option = Screen_Options::$per_page_user_option;
+
+		if ( ! is_null( $screen ) ) {
+			$screen->add_option(
+				'per_page',
+				[
+					'label'  => __( 'Number of tickets per page:', 'event-tickets' ),
+					'option' => $this->per_page_option,
+				]
+			);
+		}
+	}
+
+	/**
+	 * Returns the columns for the list table.
+	 *
+	 * @since  TBD
+	 *
+	 * @return array
+	 */
+	public function get_table_columns(): array {
+		$table_columns = [
+			'name'      => esc_html__( 'Ticket Name', 'event-tickets' ),
+			'id'        => esc_html__( 'Ticket ID', 'event-tickets' ),
+			'event'     => esc_html__( 'Event', 'event-tickets' ),
+			'start'     => esc_html__( 'Sale Starts', 'event-tickets' ),
+			'end'       => esc_html__( 'Sale Ends', 'event-tickets' ),
+			'days_left' => esc_html__( 'Days Remaining', 'event-tickets' ),
+			'price'     => esc_html__( 'Price', 'event-tickets' ),
+			'sold'      => esc_html__( 'Sold', 'event-tickets' ),
+			'remaining' => esc_html__( 'Remaining', 'event-tickets' ),
+			'sales'     => esc_html__( 'Sales', 'event-tickets' ),
+		];
+
+		/**
+		 * Filters the columns for the All Tickets Table.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $table_columns The columns for the All Tickets Table.
+		 *
+		 * @return array
+		 */
+		return apply_filters( 'tec_tickets_all_tickets_table_columns', $table_columns );
+	}
+
+	/**
+	 * Get a list of columns. The format is:
+	 * 'internal-name' => 'Title'
+	 *
+	 * @since  TBD
+	 *
+	 * @return array
+	 */
+	public function get_columns(): array {
+		return $this->get_table_columns();
+	}
+
+	/**
+	 * Get primary column for the list table.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	protected function get_primary_column_name(): string {
+		return 'name';
+	}
+
+	/**
+	 * Get hidden columns for the list table.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	public function get_hidden_columns(): array {
+		$screen = get_current_screen();
+		if ( is_null( $screen ) ) {
+			return $this->get_default_hidden_columns();
+		}
+		return get_hidden_columns( $screen );
```

Comment:
Add some space to breathe here:
```suggestion
		$screen = get_current_screen();

		if ( is_null( $screen ) ) {
			return $this->get_default_hidden_columns();
		}

		return get_hidden_columns( $screen );
```